driver/git.go

```diff
@@ -369,6 +369,15 @@ func (driver *GitDriver) writeVersion(newVersion semver.Version) (bool, error) {
 		return false, err
 	}
 
+	gitPull := exec.Command("git", "pull", "-r")
```

Comment:
I think we should only run `git pull --rebase` when `git push` hit "reject" error.